### PR TITLE
feat!: Set 21 as latest protocol version

### PR DESCRIPTION
**Pull request by perekopskiy**
## What ❔

- Updates `contracts` submodule
- Changes VM 1.4.2 according to work with updated contracts
- Changes latest protocol version to be 21

## Why ❔

Changes needed for 1.4.2 upgrade

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.
- [ ] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

File: core/lib/config/src/configs/eth_sender.rs [hunk 1]
```diff
@@ -140,8 +143,16 @@ pub struct GasAdjusterConfig {
...
-    /// Max number of l1 gas price that is allowed to be used in state keeper.
+    /// Max number of l1 gas price that is allowed to be used.
     pub max_l1_gas_price: Option<u64>,
+    /// Number of blocks collected by GasAdjuster from which `blob_base_fee` median is taken
+    #[serde(default = "GasAdjusterConfig::default_num_samples_for_blob_base_fee_estimate")]
+    pub num_samples_for_blob_base_fee_estimate: usize,
+    /// Parameter by which the pubdata fee will be multiplied for internal purposes
+    #[serde(default = "GasAdjusterConfig::default_internal_pubdata_pricing_multiplier")]
+    pub internal_pubdata_pricing_multiplier: f64,
+    /// Max blob base fee that is allowed to be used.
```
Comment (mm-zk) [inline core/lib/config/src/configs/eth_sender.rs:154]: nit: mention what happens if this is None

File: etc/env/base/contracts.toml [hunk 2]
```diff
@@ -41,7 +41,8 @@ L2_WETH_TOKEN_PROXY_ADDR="0x5E6D086F5eC079ADFF4FB3774CDf3e8D6a34F7E9"
 FRI_RECURSION_LEAF_LEVEL_VK_HASH ="0x062362cb3eaf1f631406cbe19bf2a2c5d0d9ea69d069309a6003addae9f387be"
 FRI_RECURSION_NODE_LEVEL_VK_HASH ="0x5a3ef282b21e12fe1f4438e5bb158fc5060b160559c5158c6389d62d9fe3d080"
 FRI_RECURSION_SCHEDULER_LEVEL_VK_HASH ="0xf12256f7ba1ab223ba82abb19c8366c6677596897f9e5a82016977113e52ed5d"
```
Comment (mm-zk) [inline etc/env/base/contracts.toml:43]: SCHEDULER_LEVEL_VK_HASH should also change:
FRI_RECURSION_SCHEDULER_LEVEL_VK_HASH ="0x1405880dc3317d635bddb0ab62bf5d013e5d1f462161c1f7ac3289c7fef956da"

**Review by Deniallugo — DISMISSED** [review]

File: etc/env/base/eth_sender.toml [hunk 3]
```diff
@@ -19,7 +19,7 @@ max_txs_in_flight=30 # Safe in the local environment, do not repeat on prod (rig
 proof_sending_mode="SkipEveryProof"
 
 # Max L2 blocks to commit in one L1 transaction
-max_aggregated_blocks_to_commit=10
+max_aggregated_blocks_to_commit=1
```
Comment (mm-zk) [inline etc/env/base/eth_sender.toml:22]: revert ?

**Review by mm-zk — APPROVED** [review]

File: etc/env/base/eth_sender.toml (hunk 3 above)
Comment (mm-zk) [reply etc/env/base/eth_sender.toml:22]: never mind - I misunderstood what this config does.

Comment (perekopskiy) [reply etc/env/base/eth_sender.toml:22]: L1 contracts allows max 1 L1 batch committed in a single tx

**Review by montekki — APPROVED** [review]

File: etc/env/base/eth_sender.toml [hunk 4]
```diff
@@ -19,7 +19,7 @@ max_txs_in_flight=30 # Safe in the local environment, do not repeat on prod (rig
 proof_sending_mode="SkipEveryProof"
```
Comment (koloz193) [inline etc/env/base/eth_sender.toml:20]: we're missing `PUBDATA_SENDING_MODE` , it should be set to `CALLDATA`

**Review by koloz193 — CHANGES_REQUESTED** [review]

**Comment by koloz193** [comment]
@montekki i think we also need to add something around [here](https://github.com/matter-labs/zksync-era/blob/fd6a7ee9322e2c2a809bdde128f1a71ecc0e82fa/core/lib/zksync_core/src/eth_sender/eth_tx_manager.rs#L236)
to enable the config switch to blobs
```
        let blob_gas_price = if tx.blob_sidecar.is_some() {
            Some(U256::from_big_endian(&blob_base_fee_per_gas.unwrap().to_be_bytes()))
        } else {
            None
        };

        let mut signed_tx = self
            .sign_tx(
                tx,
                base_fee_per_gas,
                priority_fee_per_gas,
                /* TODO: query adjuster */ blob_gas_price,
            )
            .await;
```

**Review by RomanBrodetski — APPROVED** [review]

**Review by mm-zk — APPROVED** [review]

**Review by popzxc — APPROVED** [review]

**Pull request merged**